### PR TITLE
BUG: `.cast()` to array outputs list instead of np.array in Pandas backend

### PR DESCRIPTION
**Pull request by timothydijamco**
# Problem

In the Pandas backend, when casting a column to an array type, a `list` is created as the output instead of a `np.array` (which is the new standard in the Pandas backend).

```
>>> series = t['arr_col'].execute()
>>> type(series[0])
numpy.ndarray
>>> series = t['arr_col'].cast('array<float64>').execute()
>>> type(series[0])
list
```

# Solution
Change the Pandas backend implementation of the cast operation to produce a `np.array`.

### CODE REVIEW

File: ibis/backends/pandas/execution/generic.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def execute_cast_series_array(op, data, type, **kwargs):
             '(e.g., number, string, or timestamp)'
         )
     return data.map(
```
Comment (icexelloss) [inline ibis/backends/pandas/execution/generic.py:114]: Can you micro if this is faster/slower?

```
pd.Series(np.array(data.to_numpy().tolist()).astype(target_dtype).tolist())
```

**Comment by icexelloss** [comment]
LGTM. Minor comment.

File: ibis/backends/pandas/execution/generic.py (hunk 1 above)
Comment (icexelloss) [reply ibis/backends/pandas/execution/generic.py:114]: The idea is that to see if putting it into a 2d array makes things faster/slower

Comment (timothydijamco) [reply ibis/backends/pandas/execution/generic.py:114]: That makes sense as an alternative to me, however it doesn't seem to work if the arrays in the Series don't all have the same length. (`np.array(data.to_numpy().tolist())` then results in nested `np.arrays` which doesn't work with `.astype(...)`)

**Comment by timothydijamco** [comment]
@gerrymanoim This PR is breaking cast-to-array on the Dask backend. I've xfailed the test that was failing. #2743 will fix this. Does that sound reasonable to you? I'd also be happy to instead create a working Dask-specific execution function for `Cast` in the meantime (currently the Dask backend borrows the Pandas backend's `Cast` execution function).

**Comment by jreback** [comment]
lgtm @timothydijamco ping when ready

File: ibis/backends/pandas/execution/generic.py (hunk 1 above)
Comment (icexelloss) [reply ibis/backends/pandas/execution/generic.py:114]: LGTM

**Review by icexelloss — APPROVED** [review]
LGTM!

**Comment by timothydijamco** [comment]
thanks, should be all set @jreback

**Comment by jreback** [comment]
thanks @timothydijamco

**Pull request merged**